src/compute/src/logging/compute.rs

```diff
@@ -890,3 +901,74 @@ impl Drop for RetractImportFrontiers {
         self.log();
     }
 }
+
+/// Extension trait to attach `ComputeEvent::DataflowError` logging operators to collections and
+/// batch streams.
+pub(crate) trait LogDataflowErrors {
+    fn log_dataflow_errors(self, logger: Logger, export_id: GlobalId) -> Self;
+}
+
+impl<G, D> LogDataflowErrors for Collection<G, D, Diff>
+where
+    G: Scope,
+    D: Data,
+{
+    fn log_dataflow_errors(self, logger: Logger, export_id: GlobalId) -> Self {
+        self.inner
+            .unary(Pipeline, "LogDataflowErrors", |_cap, _info| {
```

Comment:
Nit: Rename to `LogDataflowErrorsCollection` and (`Stream`) below so the operators have unique names.

Reply:
We often have this pattern that the same operator can be attached to different object types (streams, collections, arrangements) via an extension trait. Are you suggesting that it not a good pattern and we should move away from it?

Fine with doing that in this case, but then I wouldn't bother having traits in the mix at all and just implement free-standing `log_collection_dataflow_errors` and `log_stream_dataflow_errors` functions.

Reply:
No, that pattern is fine -- that's the purpose to having a trait here. I'm thinking more about debugging where we could pinpoint the implementation from the name, instead of having to go back to the callsite to determine which implementation we might select.

Reply:
Oh, you meant the visible name, not the trait name! Sorry :D